backend/controllers/users.js

```diff
@@ -1,33 +1,67 @@
-import { User } from '../models/user.js'
+import { User } from "../models/user.js";
+import { TokenGenerator } from "../models/token_generator.js";
 
 export const UsersController = {
   Create: async (req, res) => {
-    const existingUser = await User.exists({ email: req.body.email })
+    const existingUser = await User.exists({ email: req.body.email });
+    const token = await TokenGenerator.jsonwebtoken(req.user_id);
     if (existingUser) {
-      return res.status(400).json({ message: 'Email already in use' })
+      return res.status(400).json({ token, message: "Email already in use" });
     }
-    const newUser = new User(req.body)
+    const newUser = new User(req.body);
     newUser.save((err) => {
-      if (err) res.status(400).json({ message: err.message })
+      if (err) res.status(400).json({ token, message: err.message });
       else {
         res.status(201).json({
-          message: 'Thanks! your account has been successfully created'
-        })
+          token,
+          message: "Thanks! your account has been successfully created",
+        });
       }
-    })
+    });
+  },
+
+  Update: (req, res) => {
+    const userId = req.user_id;
+    const pet = req.body.pet;
+    if (pet === undefined || pet === null) {
+      res
+        .status(400)
+        .json({ message: "Bad request: pet is null or undefined" });
+      return;
+    }
+    User.updateOne({ _id: userId }, { pet: [pet] }, (err) => {
+      if (err) {
+        res.status(400).json({ message: "Bad request" });
+      } else {
+        res.status(200).json({ message: "OK" });
+      }
+    });
   },
 
   FindUserById: (req, res) => {
     // Extract user_id from params (Taken from :post_id in the url)
-    const user_id = req.params.user_id
+    const user_id = req.params.user_id;
     // Use user_id to find particular post
     User.findById({ _id: user_id }, async (err, user) => {
       if (err) {
-        throw err
+        throw err;
+      }
+      // const token = await TokenGenerator.jsonwebtoken(req.user_id)
+      // returns a body containing the post object and token string
+      const token = await TokenGenerator.jsonwebtoken(req.user_id);
+      res.status(200).json({ token, user });
+    });
+  },
+
+  FindUserByToken: (req, res) => {
```

Comment:
This controller function seems to be generating an auth error in postman while other routes are authenticating

Reply:
I've doubled checked this in the frontend and in postman - could your token have expired?